### DISCUSSION TITLE
Week 3: July 1-5

### DISCUSSION THREAD

**Discussion by eeholmes** [SAFS-Varanasi-Internship/Summer-2024#16] [General]
Notes

**Comment by yifeihang** [comment]
![CNN Intuitive GIF](https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/166258283/182d645a-eece-4fd7-a65a-cffb80f1738b)

**Comment by yifeihang** [comment]
07/01/24 (Mon.)
Progress:
- Used standardization to improve the model with variables 'u_wind', 'v_wind', 'sst': lower loss, a bit better prediction, prevents prediction with constant values across pixels

Goal for tomorrow:
- Work on custom loss function to calculate the loss only base on regions with values and ignore the nans.
- Compare the mean of True vs Predicted

**Comment by yifeihang** [comment]
Packages used:
xarray, numpy, dask, matplotlib, tensorflow

**Reply by ShridharS19** [reply]
+
tensorflow_decision_forests
sklearn
seaborn

**Comment by ShridharS19** [comment]
07/01/24
Progress:
- Implemented RandomForest Regression based on PCA selected features
- 
![rf3](https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/108433641/61d14266-9d8c-4dc1-8f21-831650e6c31b)
![rf4](https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/108433641/44d5cee1-c066-4419-a010-ac4c6469ec61)
![rf5](https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/108433641/2ffaabc3-b1e4-47be-9800-1cb164a51212)

To-Do
- Look at other Tree-based models and analyse dependencies between features more thoroughly

**Comment by eeholmes** [comment]
@ShridharS19 @yifeihang Try this tensorflow image and let me know if you can run your code on that:
<img width="580" alt="image" src="https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/2545978/5e439ed5-8c84-414f-a647-7b2ad4d9e79d">

**Reply by yifeihang** [reply]
Is it the one under CPU servers? If so then yes.

**Reply by eeholmes-uw** [reply]
Ah, right! I need to move that under the GPU one.

**Comment by eeholmes** [comment]
memory usage for CNN: https://datascience.stackexchange.com/questions/17286/cnn-memory-consumption

**Comment by yifeihang** [comment]
07/02/24 (Tues.)

Progress:
- Tried sereral variations on yesterday's model:
1. Custom loss function - calculate loss of regions without N/A, ignoring land
2. Change standardization to normalization
3. Change sst to topo/air_temp
None of the above improves the model, but certain functions I wrote may be handy for future alternations or evaluations
- Worked on the CUDA tutorial during model training

Goal for tomorrow:
- Use a smaller region of the image and/or less data to train, see if possible to use more variables
- Use different time ranges for training and testing (eg. Start from year _a_, train for _x_ years, and predict the following year -> vary _a_ and _x_ to see how the performance change
- Think about how to deal with N/As from the cloud blob (how to fill the data or take them out of training)
